x-pack/plugin/ent-search/qa/full-cluster-restart/build.gradle

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import org.elasticsearch.gradle.Version
+import org.elasticsearch.gradle.VersionProperties
+import org.elasticsearch.gradle.internal.info.BuildParams
+import org.elasticsearch.gradle.testclusters.StandaloneRestIntegTestTask
+
+apply plugin: 'elasticsearch.internal-java-rest-test'
+apply plugin: 'elasticsearch.bwc-test'
+
+
+dependencies {
+  testImplementation project(path: xpackModule('ent-search'))
+  javaRestTestImplementation(testArtifact(project(xpackModule('core'))))
+  javaRestTestImplementation(testArtifact(project(":qa:full-cluster-restart"), "javaRestTest"))
+}
+
+assert Version.fromString(VersionProperties.getVersions().get("elasticsearch")).getMajor() == 8:
+  "If we are targeting a branch other than 8, we should enable migration tests"
+
+BuildParams.bwcVersions.withWireCompatible(v -> v.after("8.8.0") && v.before("8.12.0")) { bwcVersion, baseName ->
```

Comment:
I understand `v.after("8.8.0")` since the entsearch module was not existing before but I think `v.before("8.12.0")` should be part of the test class. Indeed we need to make sure we will be able to test future migrations (eg. between 8.12.0 and 8.13.0)

Reply:
The best way to do this is to use `assumeTrue` as the first line of your test:

```java
public void testBehavioralAnalyticsDataRetention() throws Exception {
    assumeTrue('No need to run',  getOldClusterVersion().before(V_8_12_0))

    // ... Your test code
}
```

`assumeTrue` will cause the test to be skipped if the condition version is not met (there is also a `assumeFalse` version)

Reply:
Thanks for the feedback - changed this, but would appreciate a second set of eyes.

Reply:
I will check in the CI runs if everything behave as expected.

Reply:
Thank you for the double check.

Reply:
This is ok for me